src/random.cpp

```diff
@@ -92,9 +92,8 @@ uint32_t zmq::generate_random ()
 //    configurable via config.h
 
 //  TODO this should probably be done via config.h
-#if __cplusplus >= 201103L                                                     \
-  || (defined(__cpp_threadsafe_static_init)                                    \
-      && __cpp_threadsafe_static_init >= 200806)                               \
+#if (defined(__cpp_threadsafe_static_init)                                     \
```

Comment:
@bluca Thanks for fixing this. However, probably this is now more restrictive than necessary. While __cpp_threadsafe_static_init is defined only from gcc 8, according to https://gcc.gnu.org/projects/cxx-status.html, thread-safe initialization is implemented already from gcc 4.3. Since gcc 8 is not widely used yet, I think it would be good to query the gcc version explicitly. Do you agree? Then I would send another PR.

Reply:
The problem is that it doesn't work - it's easy to reproduce with malamute, the mlm_proto selftests (and others) always abort at shutdown as mutex lock fails with einval - see backtrace that I posted in your other PR :-/

Reply:
Note that when building with GCC 8 the problem goes away - so not sure how reliable that page is...

Reply:
the quickest test with malamute is:

```LD_LIBRARY_PATH=src/.libs src/.libs/mlm_selftest -t mlm_stream_simple```

Takes milliseconds to run and always fails when building libzmq with gcc < 8 and without this change